OpenRA.Mods.Common/ServerTraits/SkirmishLogic.cs

```diff
@@ -0,0 +1,173 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using OpenRA.Network;
+using OpenRA.Primitives;
+using OpenRA.Server;
+using OpenRA.Traits;
+using S = OpenRA.Server.Server;
+
+namespace OpenRA.Mods.Common.Server
+{
+	public class SkirmishLogic : ServerTrait, IClientJoined, INotifySyncLobbyInfo
+	{
+		class SkirmishSlot
```

Comment:
```suggestion
		readonly struct SkirmishSlot
```

Why not struct?

Reply:
oh, field loader doesn't work with structs? That should be documented. Saving structs works just fine

That's why I was getting https://github.com/OpenRA/OpenRA/pull/21206#pullrequestreview-1726306260